snuba/datasets/storages/metrics.py

```diff
@@ -57,11 +73,22 @@
         storage_set_key=StorageSetKey.METRICS,
     ),
     query_processors=[],
-    stream_loader=build_kafka_stream_loader_from_settings(
-        processor=SetsMetricsProcessor(), default_topic=Topic.METRICS,
-    ),
+    stream_loader=sets_kafka_stream_loader,
 )
 
+if settings.ENABLE_METRICS_SUBSCRIPTIONS:
+    counters_kafka_stream_loader = build_kafka_stream_loader_from_settings(
+        processor=CounterMetricsProcessor(),
+        default_topic=Topic.METRICS,
+        commit_log_topic=Topic.METRICS_COMMIT_LOG,
+        subscription_scheduler_mode=SchedulingWatermarkMode.GLOBAL,
+        subscription_scheduled_topic=Topic.SUBSCRIPTION_SCHEDULED_METRICS,
+        subscription_result_topic=Topic.SUBSCRIPTION_RESULTS_METRICS,
+    )
+else:
+    counters_kafka_stream_loader = build_kafka_stream_loader_from_settings(
+        processor=CounterMetricsProcessor(), default_topic=Topic.METRICS,
+    )
```

Comment:
Why would we want to support metrics without a commit log? This is a requirement for subscriptions